management/storageservice/client.go

```diff
@@ -84,21 +84,31 @@ func (self StorageServiceClient) GetStorageServiceByLocation(location string) (*
 	return nil, nil
 }
 
-func (self StorageServiceClient) CreateStorageService(name, location string) (*StorageService, error) {
-	if name == "" {
-		return nil, fmt.Errorf(errParamNotSpecified, "name")
+func (self StorageServiceClient) CreateAsync(parameters StorageAccountCreateParameters) (string, error) {
+	if parameters.ServiceName == "" {
+		return "", fmt.Errorf(errParamNotSpecified, "parameters.ServiceName")
```

Comment:
hehe what if I told you all these stuff are not necessary anymore. :) same is true for the entire library though. azure will just tell you "ServiceName" is missing if you leave required ones empty.

by having those guys here, we're actually duplicating the knowledge of what's required and what's not in both API service code and Library. I think this guy should be stupid and shouldn't know what's required and what's not. We expect users to get these required errors from the API and fix their mistakes. What if tomorrow Label becomes optional in a newer API version? Now you need to update the library.

Let's start with this little method here and remove all those checks.